analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -2,132 +2,110 @@
 using System.Collections.Generic;
 using System.Linq;
 
-public class EmptyNullableValueAccess
+class EmptyNullableValueAccess
 {
-    private IEnumerable<TestClass> numbers = new[]
+    private IEnumerable<TestClass> Numbers = new[]
     {
-        new TestClass { Number = 42 },
+        new TestClass() { Number = 42 },
         new TestClass(),
-        new TestClass { Number = 1 },
-        new TestClass { Number = null }
+        new TestClass() { Number = 1 },
+        new TestClass() { Number = null }
     };
 
-    public class TestClass
+    void NullAssignment()
     {
-        public int? Number { get; set; }
-    }
-
-    void Nameof(object o)
-    {
-        if (o == null)
+        int? i = null;
+        if (i.HasValue)
         {
+            Console.WriteLine(i.Value);
         }
 
-        if (o == nameof(Object))
-        {
-            o.ToString(); // Compliant
-        }
+        Console.WriteLine(i.Value); // Noncompliant {{'i' is null on at least one execution path.}}
+        //                ^
     }
 
-    public void TestNull()
-    {
-        int? i1 = null;
-        if (i1.HasValue)
-        {
-            Console.WriteLine(i1.Value);
-        }
-
-        Console.WriteLine(i1.Value); // FIXME Non-compliant {{'i1' is null on at least one execution path.}}
-        // FIXME          ~~~~~~~~
-    }
+    IEnumerable<TestClass> EnumerableExpressionWithCompilableCode() =>
+        Numbers.OrderBy(x => x.Number.HasValue).ThenBy(i => i.Number);
```

Comment:
This is out of scope right? If so, the test case should be removed or at least an appropriate comment to be added.

Reply:
Indeed, I couldn't find how the test could relate to S3655.

I have changed it to make it more relevant, and moved to a class `Linq`, which is supposed to test nullable value types use in Linq expressions.